printer/snmp.js

```diff
@@ -0,0 +1,151 @@
+
+    if (currentLevel && capacity) {
+      return (currentLevel / capacity) * 100;
+    }
+
+    return false;
+  }
+
+  /**
+   * getTonerCapacity makes a snmp query against toner capacity OID and returns value
+   * @returns Promise with toner capacity in unknown units
+   */
+  async  getTonerCapacity() {
+    return await this.executeSNMPRequest(this.printerIP, [SNMP_OBJECT_IDS.TONER_CAPACITY]);
+  }
+
+  /**
+   * getCurrentTonerLevel makes a snmp query against toner level OID and returns value
+   * @returns {Promise} Promise with current toner level in unknown units
+   */
+  async  getCurrentTonerLevel() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.CURRENT_TONER_LEVEL,
+    ]);
+  }
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  pagesPrinted() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.PAGES_PRINTED,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  macAddy() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.MAC_ADDY,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  serialNumber() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.SERIAL_NUMBER,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  modelNumber() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.MODEL_NUMBER,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  memorySize() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.MEMORY_SIZE,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async  memoryUsed() {
+    return await this.executeSNMPRequest(this.printerIP, [
+      SNMP_OBJECT_IDS.MEMORY_USED,
+    ]);
+  }
+
+  /**
+   * executreSNMPRequest makes a snmp query against the given OID
+   * @param {String} objectIdentifier OID to query
+   * @returns value corresponding to OID, or false on error
+   */
+  async executeSNMPRequest(printerIP, objectIdentifier) {
```

Comment:
you could make this function take a single parameter, and `this.printerIP` is used on line 125 instead